src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
+        capasity = defaulCapacity;
+        size = 0;
+        table = new Node[capasity];
+    }
+
+    static class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        Node(K key, V value,Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > table.length * loadFactor) {
+            resize();
+        }
+        Node<K,V> node = table[hash(key) % table.length];
+        if (node == null) {
+            table[hash(key) % table.length] = new Node<>(key,value,null);
+            size++;
+        } else {
+            while (node != null) {
 
+                if (node.next == null) {
+                    if (node.key == null) {
+                        if (key == null) {
+                            node.value = value;
+                            break;
+                        }
+                    } else if (node.key.equals(key)) {
+                        node.value = value;
+                        break;
+                    }
+                    node.next = new Node<>(key,value,null);
+                    size++;
+                    break;
+                } else if (node.key == null) {
+                    if (key == null) {
+                        node.value = value;
+                        break;
+                    }
+                } else if (node.key.equals(key)) {
+                    node.value = value;
+                    break;
+                }
+                node = node.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> node = table[hash(key) % table.length];
+        if (node != null) {
+            while (node != null) {
+                if (node.key != null) {
+                    if (node.key.equals(key)) {
+                        return node.value;
+                    }
+                } else {
+                    if (key == null) {
+                        return node.value;
+                    }
+                }
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
     }
+
+    private int hash(Object key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % capasity);
+    }
+
+    private void resize() {
+        capasity = capasity * 2;
+        Node<K, V>[] newTable = new Node[capasity];
+
+        for (int i = 0; i < table.length; i++) {
+            Node<K, V> node = table[i];
+            while (node != null) {
+                Node<K, V> next = node.next;
+                int newIndex = hash(node.key);
+                node.next = newTable[newIndex];
+                newTable[newIndex] = node;
+                node = next;
+            }
+        }
+        table = newTable;
+    }
```

Comment:
In the `resize()` method, you are directly modifying the `capacity` and `table` fields. It would be better to create a new local variable for the new capacity and a new table, and then assign them to the fields at the end of the method. This way, if there is an exception during the resizing, the original table and capacity will not be corrupted.